src/components/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,124 @@
+import { ReactComponent as IconNotebook } from './Notebook.svg';
+
+import styles from './NavBar.module.css';
+
+type NavBarProps = {
+  children?: ReactNode;
+};
+
+export default function NavBar({ children }: NavBarProps) {
+  const { pathname } = useLocation();
+
+  const menuRef = useRef<HTMLUListElement>(null);
+
+  const toogleMobileMenu = () => {
+    menuRef.current?.classList.toggle('active');
+  };
+
+  useEffect(() => {
+    const focusableElements = menuRef.current?.querySelectorAll('a, button');
+    const firstElement =
+      focusableElements && (focusableElements[0] as HTMLElement);
+    const lastElement =
+      focusableElements &&
+      (focusableElements[focusableElements.length - 1] as HTMLElement);
+
+    const focusTrap = (e: KeyboardEvent) => {
+      if (
+        window.innerWidth > 768 ||
+        !menuRef.current?.classList.contains('active') ||
+        e.key !== 'Tab'
+      )
+        return;
+
+      const isWithinMenu = menuRef.current?.contains(document.activeElement);
+
+      if (e.shiftKey) {
+        if (!isWithinMenu || document.activeElement === firstElement) {
+          e.preventDefault();
+          lastElement?.focus();
+        }
+      } else if (!isWithinMenu || document.activeElement === lastElement) {
+        e.preventDefault();
+        firstElement?.focus();
+      }
+    };
+
+    window.addEventListener('keydown', focusTrap);
+
+    return () => {
+      window.removeEventListener('keydown', focusTrap);
+    };
+  }, []);
+
+  useEffect(() => {
+    menuRef.current?.classList.remove('active');
+  }, [pathname]);
+
+  return (
+    <nav className={styles['nav-bar']}>
+      <div className={styles['mobile-menu-container']}>
+        <button
+          type="button"
+          className={styles['mobile-menu']}
+          onClick={toogleMobileMenu}
+        >
+          <IconMenu />
+        </button>
+      </div>
+      <div className={styles['home-container']}>
+        <Link
+          to="/"
+          className={styles['home-link']}
+        >
+          <IconHome className={styles['home-icon']} />
+        </Link>
+      </div>
+      <ul
+        className={styles['menu-container']}
+        ref={menuRef}
+      >
+        <div className={styles['close-button-container']}>
+          <button
+            className={styles['close-button']}
+            type="button"
+            onClick={toogleMobileMenu}
+          >
+            <IconClose />
+          </button>
+        </div>
+        <li className={`${styles['link-container']} ${styles.classes}`}>
+          <NavLink to="/">
+            <IconModules />
+            <span>Mis Clases</span>
+          </NavLink>
+        </li>
+        <li className={`${styles['link-container']} ${styles.assignments}`}>
+          <NavLink to="/assignment">
+            <IconNotebook />
+            <span>Actividades</span>
+          </NavLink>
+        </li>
+        {children &&
+          Children.map(children, (navLink, index) => (
```

Comment:
I like this! I didn't know this was possible.
I have a question about this component. For a student, we will call NavBar, and Teacher will call it with props? When I was first reading your code, I thought for a second that both Student and Teacher would have its own NavBar component according to its needs.

Reply:
The latter is correct. Our root component checks the type of user logged in our system, and based on that we will render a Student or Teacher Navbar. This Navbar is the baseline, so it only includes the options that share both roles. Later we could add more options specifically to each role.